azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,276 @@
+ * dynamic resource provider.
+ */
+interface DynamicProviderOutputs extends DynamicProviderInputs, cdnManagement.CdnManagementModels.CustomDomainsCreateResponse {
+    name: string;
+}
+
+class CDNCustomDomainResourceProvider implements pulumi.dynamic.ResourceProvider {
+    private name: string;
+
+    constructor(name: string) {
+        this.name = name;
+    }
+
+    private async getCDNManagementClient(): Promise<cdnManagement.CdnManagementClient> {
+        let clientID = azure.config.clientId;
+        let clientSecret = azure.config.clientSecret;
+        let tenantID = azure.config.tenantId;
+        let subscriptionID = azure.config.subscriptionId;
+        let credentials: ServiceClientCredentials;
+
+        // If at least one of them is empty, try looking at the env vars.
+        if (!clientID || !clientSecret || !tenantID || !subscriptionID) {
+            console.log("Checking env vars for ARM credentials.");
+            clientID = process.env["ARM_CLIENT_ID"];
+            clientSecret = process.env["ARM_CLIENT_SECRET"];
+            tenantID = process.env["ARM_TENANT_ID"];
+            subscriptionID = process.env["ARM_SUBSCRIPTION_ID"];
+        }
+
+        // If they are still empty, try to get the creds from Az CLI.
+        if (!clientID || !clientSecret || !tenantID || !subscriptionID) {
+            // `create()` will throw an error if the Az CLI is not installed or `az login` has never been run.
+            const cliCredentials = await msRestAzure.AzureCliCredentials.create();
+            subscriptionID = cliCredentials.subscriptionInfo.id;
+            credentials = cliCredentials;
+        } else {
+            credentials = await msRestAzure.loginWithServicePrincipalSecret(clientID, clientSecret, tenantID);
+        }
+
+        return new cdnManagement.CdnManagementClient(credentials, subscriptionID);
+    }
+
+    async check(olds: DynamicProviderInputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.CheckResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        let result: pulumi.dynamic.CheckResult;
+
+        // If none of the CDN properties changed, then there is nothing to be validated.
+        if (olds.profileName === news.profileName &&
+            olds.endpointName === news.endpointName &&
+            olds.customDomainHostName === news.customDomainHostName) {
+            return {
+                inputs: news
+            };
+        }
+
+        const validationOutput = await cdnClient.endpoints.validateCustomDomain(
+            news.resourceGroupName, news.profileName, news.endpointName, news.customDomainHostName);
+        if (!validationOutput.customDomainValidated) {
+            result = {
+                failures: [
+                    {
+                        property: this.name,
+                        reason: validationOutput.reason || "domain_validation_failed"
+                    }
+                ]
```

Comment:
consider trimming this down to:

```ts
            result = {
                failures: [{
                    property: this.name,
                    reason: validationOutput.reason || "domain_validation_failed",
                }],
```

(also in a few other cases).

Reply:
Done (and in a few other places)